### ISSUE TITLE
Unable to inject SpringBoot beans or use any other Bean in Tasks that doesn't implement Serializable

### ISSUE THREAD

**Issue by aXZhbg** [redisson/redisson#2318]
Hello,
I've tried running tasks Runnable/Callable that have autowired dependencies from SpringBoot and I get serialization errors on scheduling tasks or null pointer exceptions when tasks are about to execute.

I've crawled through documentation very carefully and I've even noticed there is some invalid documentation examples.

Please check out in your documentation Section 9. , 9.4.3. Distributed scheduled executor service. Scheduling a task with Spring beans.

Example code is invalid and it doesn't work.

To summarize:

Is there a way to create Callable/Runnable tasks that have SpringBoot beans as dependencies autowired - Important: that they dont have to implement Serializable interface.

Thank you.

**Comment by mrniko** [comment]
Did you register Redisson Node with provided Spring beanFactory?

**Comment by aXZhbg** [comment]
Yes I have - I have my config for RedissonClient in a .yml file like so: 

`
spring:
  redis:
    host: host
    port: 0000
    password: password
    database: 0
`
and this works, i can do methods on my redis instance with 

And then a @Configuration bean class

`
@Configuration
public class RedissonConfiguration
{

    RedissonClient redissonClient;
    BeanFactory beanFactory;

    @Autowired
    public RedissonConfiguration(RedissonClient redissonClient, BeanFactory beanFactory)
    {
        this.beanFactory = beanFactory;
        this.redissonClient = redissonClient;
    }

    @Bean
    public void setup()
    {
        RedissonNodeConfig nodeConfig = new RedissonNodeConfig(redissonClient.getConfig());
        nodeConfig.setBeanFactory(beanFactory);
        nodeConfig.setExecutorServiceWorkers(Collections.singletonMap("myExecutor", 1));
        RedissonNode node = RedissonNode.create(nodeConfig);
        node.start();
    }

}
`

And my task class where i try to autowire like shown in documentation

`
public class MyTask implements Runnable, Serializable
{

    @Autowired
    MyBean myBean;


    public void run()
    {
        myBean.doSomething();
    }

}
`
As you can see i put in implements Serializable as well, it doesnt have that in documentation, ive tried every possible codec in documentation, and if it doesnt have Serializable it just throws without even persisting the job.

Another problem with this is that myBean is not autowired on execution - i get a null pointer exception.

Also - if i try to just put it in a constructor like so:
`
public class MyTask implements Runnable, Serializable
{
    
    MyBean myBean;

    public MyTask(MyBean myBean)
    {
        this.myBean = myBean;
    }
    

    public void run()
    {
        myBean.doSomething();
    }

}
`
I get an error saying MyBean has to implement Serializable, which means that i cannot have any outside dependencies injected into job. For example I would like to send a RPC method call which is not possible since i cannot put Serializable on my dependencies

Is there a solution to this ?

Thank you

**Comment by mrniko** [comment]
Unable to reproduce it. Please find my example below. MyBean shouldn't have serializable interface. 

[springboot.zip](https://github.com/redisson/redisson/files/3604368/springboot.zip)

**Comment by aXZhbg** [comment]
bump

**Comment by mrniko** [comment]
Still have errors caused by spring configuration errors:

Caused by: org.springframework.context.ApplicationContextException: Unable to start ServletWebServerApplicationContext due to missing ServletWebServerFactory bean.

Caused by: org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'com.redisson.demo.SampleBean' available: expected at least 1 bean which qualifies as autowire candidate.

**Comment by jeff-weng** [comment]
Hi @aXZhbg, was wondering if you ever figured out how to fix this issue as I am running into the same problem

**Comment by aXZhbg** [comment]
> Hi @aXZhbg, was wondering if you ever figured out how to fix this issue as I am running into the same problem

No I haven't, the zipped project mrniko posted earlier in issue thread doesn't work either.
Since the feature I needed were jobs I gave up on redisson and moved on to quartz.

**Comment by newGaoXin** [comment]
I also have this problem, which is caused by the introduction of spring-dev-tools. If you also use it in the project, try to set the dependent <scope> to runtime, and you can get the beans in the container normally after compiling and running.
If you want to run locally through idea, you can try to comment out the spring-dev-tools dependency

**Comment by CarryWll** [comment]
> I also have this problem, which is caused by the introduction of spring-dev-tools. If you also use it in the project, try to set the dependent to runtime, and you can get the beans in the container normally after compiling and running. If you want to run locally through idea, you can try to comment out the spring-dev-tools dependency

Hello, would you like to ask if there are any other operations? Can I comment out the spring-dev-tools or not

**Comment by newGaoXin** [comment]
no other operations，you can try&nbsp;comment out the spring-dev-tools&nbsp;




------------------&nbsp;Original&nbsp;------------------
From: ***@***.***&gt;; 
Date: 2023年11月9日(星期四) 中午11:33
To: ***@***.***&gt;; 
Cc: ***@***.***&gt;; ***@***.***&gt;; 
Subject: Re: [redisson/redisson] Unable to inject SpringBoot beans or use any other Bean in Tasks that doesn&#39;t implement Serializable (#2318)





  
I also have this problem, which is caused by the introduction of spring-dev-tools. If you also use it in the project, try to set the dependent to runtime, and you can get the beans in the container normally after compiling and running. If you want to run locally through idea, you can try to comment out the spring-dev-tools dependency
  
Hello, would you like to ask if there are any other operations? Can I comment out the spring-dev-tools or not

**Comment by CarryWll** [comment]
> no other operations，you can try&nbsp;comment out the spring-dev-tools&nbsp;
> […](#)
> ------------------&nbsp;Original&nbsp;------------------ From: ***@***.***&gt;; Date: 2023年11月9日(星期四) 中午11:33 To: ***@***.***&gt;; Cc: ***@***.***&gt;; ***@***.***&gt;; Subject: Re: [redisson/redisson] Unable to inject SpringBoot beans or use any other Bean in Tasks that doesn&#39;t implement Serializable (#2318) I also have this problem, which is caused by the introduction of spring-dev-tools. If you also use it in the project, try to set the dependent to runtime, and you can get the beans in the container normally after compiling and running. If you want to run locally through idea, you can try to comment out the spring-dev-tools dependency Hello, would you like to ask if there are any other operations? Can I comment out the spring-dev-tools or not — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: ***@***.***&gt;



> no other operations，you can try&nbsp;comment out the spring-dev-tools&nbsp;
> […](#)
> ------------------&nbsp;Original&nbsp;------------------ From: ***@***.***&gt;; Date: 2023年11月9日(星期四) 中午11:33 To: ***@***.***&gt;; Cc: ***@***.***&gt;; ***@***.***&gt;; Subject: Re: [redisson/redisson] Unable to inject SpringBoot beans or use any other Bean in Tasks that doesn&#39;t implement Serializable (#2318) I also have this problem, which is caused by the introduction of spring-dev-tools. If you also use it in the project, try to set the dependent to runtime, and you can get the beans in the container normally after compiling and running. If you want to run locally through idea, you can try to comment out the spring-dev-tools dependency Hello, would you like to ask if there are any other operations? Can I comment out the spring-dev-tools or not — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you commented.Message ID: ***@***.***&gt;

Thank you for your help, I have successfully achieved
